searchlib/src/vespa/searchlib/common/bitvector.cpp

```diff
@@ -199,23 +208,36 @@ BitVector::repairEnds()
 void
 BitVector::andWith(const BitVector & right)
 {
-    verifyContains(*this, right);
+    verifyInclusiveStart(*this, right);
 
-    IAccelrated::getAccelrator().andBit(getActiveStart(), right.getWordIndex(getStartIndex()), getActiveBytes());
+    uint32_t commonBytes = std::min(getActiveBytes(), numActiveBytes(getStartIndex(), right.size()));
+    IAccelrated::getAccelrator().andBit(getActiveStart(), right.getWordIndex(getStartIndex()), commonBytes);
+    if (right.size() < size()) {
+        clearInterval(right.size(), size());
+    }
 
-    setGuardBit();
+    repairEnds();
```

Comment:
Note that repairEnds() fails to set guard bit if size() is 0.

Reply:
Not guarding setGuardBit.